plugins/parcelize/parcelize-compiler/parcelize.backend/src/org/jetbrains/kotlin/parcelize/AndroidSymbols.kt

```diff
@@ -185,6 +186,12 @@ class AndroidSymbols(
         }
     }.symbol
 
+    val kotlinTimeDuration: IrClassSymbol = createClass(
+        kotlinTime, "Duration", ClassKind.CLASS, Modality.FINAL, true
+    ).apply {
+        owner.valueClassRepresentation = InlineClassRepresentation(Name.identifier("rawValue"), irBuiltIns.longType as IrSimpleType)
```

Comment:
Isn't it the reliance on internal implementation details of `Duration`, i.e. the name of the backing field?

Reply:
Thanks for checking!

Parcelize never uses the name for anything. The only thing that it uses is the underlying type itself (long in this case). So, the name is only here because we need to have one for an `InlineClassRepresentation` object. The actual name that we use here can be anything and doesn't have to match the `Duration` source.